resilience4j-ratelimiter/src/main/java/io/github/resilience4j/ratelimiter/internal/AtomicRateLimiter.java

```diff
@@ -100,15 +101,30 @@ public void changeLimitForPeriod(final int limitForPeriod) {
         ));
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void changeLimitRefreshPeriod(Duration refreshPeriod) {
+        RateLimiterConfig newConfig = RateLimiterConfig.from(state.get().config)
+            .limitRefreshPeriod(refreshPeriod)
+            .build();
+        nanoTimeStart.compareAndSet(nanoTimeStart.get(), nanoTime());
+        state.updateAndGet(currentState -> new State(
```

Comment:
These two consecutive atomic operations aren't actually atomic on their own, and this can lead to inconsistencies of calculations that happen between these operations. I'd try to avoid changing nanoTimeStart altogether.

Reply:
I think the nanoTimeStart is part of the state.
wdyt